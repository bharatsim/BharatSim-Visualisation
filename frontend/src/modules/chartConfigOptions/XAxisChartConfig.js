import React from 'react';
import PropTypes from 'prop-types';

import { Box, fade, makeStyles, Typography } from '@material-ui/core';
import Dropdown from '../../uiComponent/Dropdown';
import { convertObjectArrayToOptionStructure } from '../../utils/helper';

const useStyles = makeStyles((theme) => {
  return {
    fieldContainer: {
      display: 'flex',
      justifyContent: 'space-between',
      padding: theme.spacing(2),
      borderColor: fade(theme.colors.primaryColorScale['500'], 0.24),
      border: '1px solid',
      borderRadius: theme.spacing(1),
      marginBottom: theme.spacing(4),
      '&:last-child': {
        marginBottom: 0,
      },
    },
  };
});

function XAxisChartConfig({ headers, updateConfigState, configKey, error, value }) {
  const classes = useStyles();
  function handleXChange(selectedValue) {
    updateConfigState(configKey, selectedValue);
  }

  return (
    <Box>
      <Box mb={1} pl={2}>
        <Typography variant="subtitle2"> X-axis</Typography>
      </Box>
      <Box className={classes.fieldContainer}>
        <Dropdown
          options={convertObjectArrayToOptionStructure(headers, 'name', 'name')}
          onChange={handleXChange}
          id="dropdown-x"
          label="select x axis"
          error={error}
          value={value}
        />
      </Box>
    </Box>
  );
}

XAxisChartConfig.defaultProps = {
  error: '',
  value: '',
};

XAxisChartConfig.propTypes = {
  headers: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
    }),
  ).isRequired,
  updateConfigState: PropTypes.func.isRequired,
  configKey: PropTypes.string.isRequired,
  error: PropTypes.string,
  value: PropTypes.string,
};

export default XAxisChartConfig;
