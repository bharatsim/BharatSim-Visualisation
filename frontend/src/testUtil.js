// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent, within } from '@testing-library/react';

export function selectDropDownOption(container, dropDownId, optionId) {
  const dropDown = container.getByTestId(dropDownId);
  fireEvent.mouseDown(within(dropDown).getByRole('button'));
  const options = within(
    within(document.getElementById(`menu-${dropDownId}`)).getByRole('listbox'),
  );
  fireEvent.click(options.getByTestId(`${dropDownId}-${optionId}`));
}
