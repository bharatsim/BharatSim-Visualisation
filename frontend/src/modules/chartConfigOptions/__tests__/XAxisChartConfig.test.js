import React from 'react';
import { render } from '@testing-library/react';

import XAxisChartConfig from '../XAxisChartConfig';
import { selectDropDownOption } from '../../../testUtil';

describe('<XAxisChartConfig />', () => {
  const props = {
    headers: [
      { name: 'a', type: 'number' },
      { name: 'b', type: 'number' },
      { name: 'c', type: 'number' },
    ],
    updateConfigState: jest.fn(),
    configKey: 'xAxis',
    error: '',
    value: '',
  };
  it('should match snapshot', () => {
    const { container } = render(<XAxisChartConfig {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should call setConfig callback after value change', () => {
    const renderedContainer = render(<XAxisChartConfig {...props} />);

    selectDropDownOption(renderedContainer, 'dropdown-x', 'a');

    expect(props.updateConfigState).toHaveBeenCalledWith('xAxis', 'a');
  });
});
