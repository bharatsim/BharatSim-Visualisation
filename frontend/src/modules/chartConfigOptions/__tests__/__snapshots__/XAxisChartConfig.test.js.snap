// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XAxisChartConfig /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2"
      >
         X-axis
      </h6>
    </div>
    <div
      class="MuiBox-root MuiBox-root-4 makeStyles-fieldContainer-1"
    >
      <div
        class="MuiFormControl-root makeStyles-formControl-5"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
          data-shrink="false"
          id="dropdown-label"
        >
          select x axis
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
          data-testid="dropdown-x"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="dropdown-label dropdown-x"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
            id="dropdown-x"
            role="button"
            tabindex="0"
          >
            <span>
              â€‹
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
